crates/dslab-core/src/context.rs

```diff
@@ -650,4 +663,505 @@ impl SimulationContext {
+        ///
+        /// # Examples:
+        /// ```rust
+        ///
+        /// use std::rc::Rc;
+        /// use std::cell::RefCell;
+        ///
+        /// use serde::Serialize;
+        ///
+        /// use dslab_core::{cast, Simulation, SimulationContext, Event, EventHandler};
+        ///
+        /// struct Client {
+        ///     ctx: SimulationContext,
+        /// }
+        ///
+        /// #[derive(Clone, Serialize)]
+        /// struct Start {
+        ///     jobs: u32,
+        /// }
+        ///
+        /// impl Client {
+        ///     fn on_start(&self, jobs: u32) {
+        ///        for i in 1..=jobs {
+        ///             self.ctx.spawn(self.step_waiting(i));
+        ///        }
+        ///     }
+        ///
+        ///     async fn step_waiting(&self, num_steps: u32) {
+        ///         for _i in 0..num_steps {
+        ///             self.ctx.async_wait_for(1.).await;
+        ///         }
+        ///     }
+        /// }
+        ///
+        /// impl EventHandler for Client {
+        ///     fn on(&mut self, event: Event) {
+        ///         cast!(match event.data {
+        ///             Start { jobs } => {
+        ///                 self.on_start(jobs);
+        ///             }
+        ///         })
+        ///     }
+        /// }
+        ///
+        ///
+        /// let mut sim = Simulation::new(42);
+        ///
+        /// let client_ctx = sim.create_context("client");
+        /// let client_id = client_ctx.id();
+        /// let client = Rc::new(RefCell::new(Client {ctx: client_ctx }));
+        ///
+        /// sim.add_handler("client", client);
+        ///
+        /// let root_ctx = sim.create_context("root");
+        /// root_ctx.emit(Start{ jobs: 10 }, client_id, 10.);
+        ///
+        /// sim.step_until_no_events();
+        ///
+        /// assert_eq!(sim.time(), 20.);
+        /// ```
+        ///
+        /// ```should_panic
+        /// use dslab_core::{Simulation, SimulationContext};
+        ///
+        /// struct Client {
+        ///     ctx: SimulationContext,
+        /// }
+        ///
+        /// impl Client {
+        ///     fn start(&self, jobs: u32) {
+        ///        for i in 1..=jobs {
+        ///             self.ctx.spawn(self.step_waiting(i));
+        ///        }
+        ///     }
+        ///
+        ///     async fn step_waiting(&self, num_steps: u32) {
+        ///         for _i in 0..num_steps {
+        ///             self.ctx.async_wait_for(1.).await;
+        ///         }
+        ///     }
+        /// }
+        ///
+        /// let mut sim = Simulation::new(42);
+        /// let mut client = Client { ctx: sim.create_context("client") };
+        ///
+        /// // panics because spawning tasks via SimulationContext
+        /// // without being registered is prohibited due to security reasons.
+        /// // Register Client via Simulation::add_handler as in previous example
```

Comment:
```suggestion
        /// // Panics because spawning async tasks by a component without event handler
        /// // is prohibited due to safety reasons.
        /// // Register Client via Simulation::add_handler as in the previous example.
```

Reply:
Там же safety имелась ввиду?